akka-http-core/src/main/java/akka/http/javadsl/model/HttpMessage.java

```diff
@@ -55,6 +58,11 @@
      */
     ResponseEntity entity();
 
+    /**
+     *  Drains entity stream of this message
+     */
+    CompletionStage<Done> ignoreDataBytesJ(Materializer materializer);
```

Comment:
@ktoso I pretty much like declaring result explicitly as proposed in b), but the issue here might be following. Most probably `discardDataBytes()` will return some intermediate class (`EntityDiscarder`?) and only calling `.future` (which requires implicit materializer) or `.completionStage` (which requires explicit one) will do the job. The problem is that user is not forced to invoke method on discarder, and might expect that after invoking `.discardDataBytes()` data stream is ignored.

Or do you want to perform actual job in `discardDataBytes()` and then possibly convert between Future/CompletionStage? (Is that even possible taking differences between scala/java API on streams into account?)